add 2 functions convertUserGuessToArray getColorsToGuess

adimalka14 opened the pull request.

adimalka14 committed:
add 2 functions convertUserGuessToArray getColorsToGuess

ItamarColler commented on src/game-status/adiMalka.js:
```diff
@@ -0,0 +1,28 @@
+const optionToGuess = 4;
+
+/**
+ * thr function get array of strung - colors and return random array of new options
```
the function get an array of strings -

ItamarColler replied:
the function gets an array of colors (strings) and generates an array that the user needs to guess, right? 
if that's the case, just retype the description : the function gets an array of strings (colors) and returns random array of colors for the user to guess

ItamarColler commented on src/game-status/adiMalka.js:
```diff
@@ -0,0 +1,28 @@
+}
+
+/**
+ * thr function get array of strung - colors and return random array of new options
```
change the description to align with the function. it should should explain the convertion of the array

ItamarColler commented on src/game-status/adiMalka.js:
```diff
@@ -0,0 +1,28 @@
+function convertUserGuessToArray(userInput,colorOptions ){
+    const numbersInput = userInput.trim().replace(/\s+/g, ' ').split(' ');
+
+    if(numbersInput.length !== optionToGuess ||
```
create a function that validates the input. even create a utils folder and a file for it.

adimalka14 committed:
Merge branch 'refs/heads/main' into adi-malka

adimalka14 committed:
fix comments.
split convertUserGuessToArray
to 3 helper functions :
removeRedundantSpaces,
validateUserGuess,
mapUserInputToColors.

add to consts file OPTION_TO_GUESS
rename OPTIONS to COLORS_OPTIONS

adimalka14 committed:
fix comment in
getColorsToGuess

ItamarColler commented:
great explanations! you can merge the PR 👍
